gocourse07/aviary.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Sensor interface {
+	measure() Data
+}
+
+type TemperatureSensor struct {
+	ID          int
+	Temperature int
+}
+
+func (t TemperatureSensor) measure() Data {
+	return NewData(t.ID, t.Temperature)
+}
+
+type LightSensor struct {
+	ID    int
+	Light int
+}
+
+func (l LightSensor) measure() Data {
+	return NewData(l.ID, l.Light)
+}
+
+type WetnessSensor struct {
+	ID      int
+	Wetness int
+}
+
+func (w WetnessSensor) measure() Data {
+	return NewData(w.ID, w.Wetness)
+}
+
+type Data struct {
+	SensorID    int
+	Measurement int
+}
+
+func NewData(ID int, measurement int) Data {
```

Comment:
Параметри в функціях завжди з маленької:
```suggestion
func NewData(id int, measurement int) Data {
```